app/src/main/java/com/github/se/gatherspot/ui/eventUI/EventDataForm.kt

```diff
@@ -608,3 +643,122 @@ enum class EventAction {
   CREATE,
   EDIT
 }
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MyDatePickerDialog(onDateSelected: (String) -> Unit, onDismiss: () -> Unit) {
+  fun convertMillisToDate(millis: Long): String {
+    val formatter =
+        SimpleDateFormat(EventFirebaseConnection.DATE_FORMAT_DISPLAYED, Locale.getDefault())
+    return formatter.format(Date(millis))
+  }
+  val datePickerState = rememberDatePickerState()
+
+  val selectedDate = datePickerState.selectedDateMillis?.let { convertMillisToDate(it) } ?: ""
+
+  DatePickerDialog(
+      onDismissRequest = { onDismiss() },
+      confirmButton = {
+        Button(
+            onClick = {
+              onDateSelected(selectedDate)
+              onDismiss()
+            }) {
+              Text(text = "OK")
+            }
+      },
+      dismissButton = { Button(onClick = { onDismiss() }) { Text(text = "Cancel") } }) {
+        DatePicker(state = datePickerState)
+      }
+}
+
+@Composable
+fun MyDatePickerDialog(onDateChange: (String) -> Unit, testTag: String) {
+
+  var showDatePicker by remember { mutableStateOf(false) }
+
+  androidx.compose.material.IconButton(
+      onClick = {
+        // Open DatePickerDialog
+        showDatePicker = true
+      },
+      modifier = Modifier.height(HEIGHT).testTag("DatePickerButton$testTag"),
+  ) {
+    androidx.compose.material.Icon(
+        modifier = Modifier.size(HEIGHT.times(0.5f)).testTag("DatePickerIcon$testTag"),
+        painter = rememberVectorPainter(image = Icons.Filled.DateRange),
+        contentDescription = "Date picker icon")
+  }
+
+  if (showDatePicker) {
+    MyDatePickerDialog(onDateSelected = { onDateChange(it) }) { showDatePicker = false }
+  }
+}
+
+@Composable
+fun MyTimePickerDialog(
+    onTimeSelected: (String) -> Unit,
+    onDismiss: () -> Unit,
+    initialTime: String,
+    title: String
+) {
+
+  val listener =
+      TimePickerDialog.OnTimeSetListener { _, hour, minute ->
+        val time = (hour.toString().padStart(2, '0') + ":" + minute.toString().padStart(2, '0'))
+        onTimeSelected(time)
+        onDismiss()
+      }
+
+  val initialHour =
+      try {
+        initialTime.split(":", limit = 2)[0].toInt()
+      } catch (e: Exception) {
+        12
```

Comment:
I think it's not a good idea to let theses magic values in the code. This is not very explicit in my opinion

Reply:
no that's just how to retrieve the hour from a string representing a time. The 2 says that we want to split the string representing time in 2 substring and 0 that we want the first the first of these substring.
As for the 12 it's an arbitrary value yes, but it's just for opening the time Picker dialog. So the clock starts on 12 : 00. 

Overall even the value of initialTime doesn't really matter, the user gets to set the clock how they please